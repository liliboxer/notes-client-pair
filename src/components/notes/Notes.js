import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Notes({ notes }) {
  const notesList = notes.map(({ title, _id }) => (
    <li key={_id}>
      <Link to={`/notes/${_id}`}>{title}</Link>
    </li>
  ));
  return <ul>{notesList}</ul>;
}

Notes.propTypes = {
  notes: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Notes;
