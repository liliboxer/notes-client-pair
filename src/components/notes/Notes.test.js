import React from 'react';
import { shallow } from 'enzyme';
import Notes from './Notes';

describe('Notes component', () => {
  it('renders Notes', () => {
    const wrapper = shallow(<Notes notes={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
