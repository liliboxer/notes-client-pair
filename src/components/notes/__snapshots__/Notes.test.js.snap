// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes component renders Notes 1`] = `<ul />`;
